### PR TITLE
Fix android so cursor doesn't reset to position zero

### LINKED ISSUE #2201: [Bug] Typing in an entry sets the cursor position to 0 after each character
### Description

Each text change (add and remove) the cursor is reset to the start of the text box.

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###

On android when you set the native text it causes the cursor to reset to position 0. 

When entering text into the native field it would cause

- fixes: #2201

### PR Checklist ###

- [ ] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [ ] Targets a single property for a single control (or intertwined few properties)
- [ ] Adds the property to the appropriate interface
- [ ] Avoids any changes not essential to the handler property
- [ ] Adds the mapping to the PropertyMapper in the handler
- [ ] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [ ] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [ ] Tags ported renderer methods with [PortHandler]
- [ ] Adds an example of the property to the sample project (MainPage)
- [ ] Adds the property to the stub class
- [ ] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might affect accessibility?
- [ ] Does this PR introduce a new control? (If yes, add an example using SemanticProperties to the SemanticsPage)
- [ ] APIs that modify focusability?
- [ ] APIs that modify any text property on a control?
- [ ] Does this PR modify view nesting or view arrangement in anyway?
- [ ] Is there the smallest possibility that your PR will change accessibility? 
- [ ] I'm not sure, please help me

If any of the above checkboxes apply to your PR, then the PR will need to provide testing to demonstrate that accessibility still works.

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -15,6 +15,12 @@
     <AndroidUseIntermediateDesignerFile>false</AndroidUseIntermediateDesignerFile>
   </PropertyGroup>
 
+  <PropertyGroup Condition="$(TargetFramework.StartsWith('MonoAndroid'))">
```
Comment (jonathanpeppers) [inline Directory.Build.targets:18]: I think you should put this on the previous `<PropertyGroup/>`:
```
Condition="($(TargetFramework.StartsWith('MonoAndroid')) or '$(TargetPlatformIdentifier)' == 'Android') and '$(AndroidApplication)' != 'true'"
```
What you have here would prevent `Resource.designer.cs` from being created in "app" projects, and I think you want to keep those?

Comment (PureWeen) [reply Directory.Build.targets]: 👍

**Comment by rmarinho** [comment]
Test is failing @PureWeen

**Review by Redth — APPROVED** [review]

File: Directory.Build.targets [hunk 2]
```diff
@@ -10,10 +10,11 @@
     Comment out this section if you need to update Resource.designer.cs files.
     See DEVELOPMENT.md#Android for details.
   -->
-  <PropertyGroup Condition="'$(TargetPlatformIdentifier)' == 'Android' and '$(AndroidApplication)' != 'true'">
+  <PropertyGroup Condition="($(TargetFramework.StartsWith('MonoAndroid')) or '$(TargetPlatformIdentifier)' == 'Android') and '$(AndroidApplication)' != 'true'">
```
Comment (mattleibow) [inline Directory.Build.targets:13]: Is this supported in MonoAndroid?

**Review by mattleibow — APPROVED** [review]

File: Directory.Build.targets (hunk 2 above)
Comment (PureWeen) [reply Directory.Build.targets:13]: Yes, once I added this it started compiling for me locally

**Comment by UkeHa** [comment]
@PureWeen, if you use a behavior to an entry and add something to the entry (e.g. / or . or - in a date) it resets the cursor position back to the position of the first time you've added something on Android. It works correctly in iOS.

**Pull request merged**